course/src/main/java/in/testpress/course/util/ExoPlayerUtil.java

```diff
@@ -247,13 +231,97 @@ public ExoPlayerUtil(Activity activity, FrameLayout exoPlayerMainFrame, String u
+            return;
+        }
+
+        String watermarkType = session.getInstituteSettings().getVideoWatermarkType();
+        isDynamic = watermarkType.equals("Dynamic");
+        if (!watermarkType.equals("Dynamic") && !watermarkType.equals("Static")) {
+            return;
+        }
+
+        ProfileDetails profileDetails = TestpressUserDetails.getInstance().getProfileDetails();
+
+        if (profileDetails == null) {
+            handleProfileLoading();
+        } else {
+            addWatermarkOverlay(profileDetails);
+        }
+    }
+
+    private void handleProfileLoading() {
+        TestpressUserDetails.getInstance().load(activity, new TestpressCallback<ProfileDetails>() {
+            @Override
+            public void onSuccess(ProfileDetails userDetails) {
+                addWatermarkOverlay(userDetails);
+            }
+
+            @Override
+            public void onException(TestpressException exception) {
+                Log.e("Watermark", "Failed to load user details", exception);
+            }
+        });
+    }
+
+    private void addWatermarkOverlay(ProfileDetails profileDetails) {
+        String overlayText = (profileDetails.getEmail() != null && !profileDetails.getEmail().isEmpty())
+                ? profileDetails.getEmail()
+                : profileDetails.getUsername();
+
+        FrameLayout parentLayout = getOrCreateParentLayout();
+        WatermarkOverlay watermark = createWatermark(overlayText);
+
+        parentLayout.addView(watermark);
+    }
+
+    private FrameLayout getOrCreateParentLayout() {
+        FrameLayout parent;
+
+        if (exoPlayerLayout.getParent() instanceof FrameLayout) {
+            Log.d("TAG", "Using existing parent layout");
+            parent = (FrameLayout) exoPlayerLayout.getParent();
+        } else {
+            Log.d("TAG", "Creating new parent layout");
+            parent = new FrameLayout(activity);
+            ViewGroup.LayoutParams layoutParams = exoPlayerLayout.getLayoutParams();
+            parent.setLayoutParams(layoutParams);
+
+            exoPlayerLayout.setLayoutParams(new FrameLayout.LayoutParams(
+                    FrameLayout.LayoutParams.MATCH_PARENT,
+                    FrameLayout.LayoutParams.MATCH_PARENT
+            ));
+
+            parent.addView(exoPlayerLayout);
+        }
+
+        return parent;
+    }
+
+    private WatermarkOverlay createWatermark(String text) {
+        WatermarkOverlay watermark = new WatermarkOverlay(activity);
+        watermark.setWatermark(text);
+
+        if (isDynamic) {
+            watermark.setDynamicWatermark();
+        } else {
+            String position = "top-right";
+            if (session == null || session.getInstituteSettings().getVideoWatermarkPosition() == null) {
+                position = "top-right";
+            } else {
+                position = session.getInstituteSettings().getVideoWatermarkPosition();
+            }
```

Comment:
```suggestion
            if (session != null && session.getInstituteSettings().getVideoWatermarkPosition() != null) {
                position = session.getInstituteSettings().getVideoWatermarkPosition();
            }
```

Even more simpler option is 

```kotlin
getVideoWatermarkPosition() {
    return this.videoWaterMarkPosition or "top-right";
}
```